### PR TITLE
Indexing, Showing, Deleting and Updating Users

### PR DESCRIPTION
**Pull request by brokenheart1712**

### CODE REVIEW

File: app/views/users/_user.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<p class="users">
+  <% @users.each do |user| %>
+  <p> <%= link_to user.name, user %>
+    <% if current_user.level == 1 && !current_user?(user) %>
+      | <%= link_to "Delete", user, method: :delete,
+                    data: {confirm: I18n.t(:are_you_sure)} %>
+    <% end %>
+  </p>
+<% end %>
+</p>
```
Comment (NguyenTanDuc) [inline app/views/users/_user.html.erb]: Add new line

File: db/seeds.rb [hunk 2]
```diff
@@ -14,5 +14,19 @@
 words = Word.order(:created_at).take(50)
 4.times do |n|
   content = "Vietnamese"
-  words.each {|word| word.answers.create! content: content, correct: n == 1}
+  words.each { |word| word.answers.create! content: content, correct: n == 1 }
```
Comment (NguyenTanDuc) [inline db/seeds.rb]: remove space {  }

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,6 +1,14 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
+
+  def index
+    @users = User.paginate(page: params[:page])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,6 +1,14 @@
 class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :admin_user, only: :destroy
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: write in admin namespace

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -16,10 +24,47 @@ def create
...
+  def logged_in_user
+    unless logged_in?
+      store_location
+      flash[:danger] = I18n.t(:please_log_in)
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
+    @user = User.find(params[:id])
```
Comment (chuanhtuan) [inline app/controllers/users_controller.rb]: ()

File: app/models/user.rb [hunk 6]
```diff
@@ -1,47 +1,46 @@
...
 
   def User.new_token
     SecureRandom.urlsafe_base64
   end
-  
+
   def remember
     self.remember_token = User.new_token
-    update_attribute :remember_digest, User.digest(remember_token)  
+    update_attribute :remember_digest, User.digest(remember_token)
```
Comment (chuanhtuan) [inline app/models/user.rb]: use update_attributes instead of update_attribute

File: app/models/user.rb [hunk 7]
```diff
@@ -1,47 +1,46 @@
...
+  def authenticated? remember_token
     return false if remember_digest.nil?
-    BCrypt::Password.new remember_digest.is_password? remember_token 
+    BCrypt::Password.new remember_digest.is_password? remember_token
   end
-  
+
   def forget
-    update_attribute :remember_digest, nil 
+    update_attribute :remember_digest, nil
```
Comment (chuanhtuan) [inline app/models/user.rb]: use update_attributes instead of update_attribute

File: app/views/users/_user.html.erb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<p class="users">
+  <% @users.each do |user| %>
+  <p> <%= link_to user.name, user %>
+    <% if current_user.level == 1 && !current_user?(user) %>
+      | <%= link_to "Delete", user, method: :delete,
```
Comment (chuanhtuan) [inline app/views/users/_user.html.erb]: admin namespace

**Pull request closed without merging**